modular_zubbers/modules/plexagon_selfserve/code/plexagon_selfserve.dm

```diff
@@ -74,6 +74,10 @@
 		if(tgui_alert(usr, "You are a member of security and/or command, make sure that you ahelp before punching out! If you decide to punch back in later, you will need to go to the Head of Personnel or Head of Security. Do you wish to continue?", "[src]", list("No", "Yes")) != "Yes")
 			return FALSE
 
+	if(istype(authenticated_card.trim, /datum/id_trim/job/prisoner))
+		if(tgui_alert(usr, "You cannot clock out of prison. Nice try, inmate."))
```

Comment:
```suggestion
		tgui_alert(usr, "You cannot clock out of prison. Nice try, inmate.")
```
This should not be an if statement, otherwise they can still clock out by cancelling the dialog box.